server/src/transactions/entities/transaction.entity.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  OneToMany,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm';
+import { TransactionStatus } from '../transaction-status.enum';
+import { User } from '../../users/entities/user.entity';
+import { Asset } from '../../assets/entities/asset.entity';
+import { Organization } from '../../organizations/entities/organization.entity';
+
+@Entity('transactions')
+export class Transaction {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({
+    type: 'enum',
+    enum: TransactionStatus,
+    default: TransactionStatus.IN_PROGRESS,
+  })
+  status: TransactionStatus;
+
+  @Column()
+  created_date: Date;
+
+  @ManyToOne((_type) => User, (user) => user.transactions, { eager: false })
+  @JoinColumn()
+  donater_user: User;
+
+  @ManyToOne(
+    (_type) => Organization,
+    (organization) => organization.transactions,
+    { eager: false },
+  )
+  @JoinColumn()
+  donater_organization?: Organization;
+
+  @ManyToOne((_type) => Asset, (asset) => asset.transactions, { eager: false })
+  asset: Asset;
```

Comment:
i think the `JoinColumn` should be here no?
asset_id is in the transactions table, so this table owns the relationship

Reply:
so remove corresponding from asset entity